pkg/cmd/server/command.go

```diff
@@ -126,4 +129,18 @@ func (cfg *Config) Complete(args []string) {
 		// if we've explicitly called out an etcd server, don't start etcd in-process
 		cfg.StartEtcd = !cfg.EtcdAddr.Provided
 	}
+
+	// if this is an all-in-one start, be sure to add our hostname to the NodeList if it is not already present
+	isAllInOne := (len(args) == 0)
+	if isAllInOne {
+		nodeList := util.NewStringSet(cfg.NodeList...)
```

Comment:
build the set by iterating over the list, `ToLower`-ing, and `ToLower` hostname, to make sure `--hostname=Localhost --nodes=localhost` doesn't make you end up with `localhost, localhost`